src/Microsoft.ApiDesignGuidelines.Analyzers/UnitTests/DoNotDeclareStaticMembersOnGenericTypesTests.cs

```diff
@@ -12,12 +12,268 @@ public class DoNotDeclareStaticMembersOnGenericTypesTests : DiagnosticAnalyzerTe
+    {
+        Console.Write(data);
+    }
+
+    protected internal static void OutputProtectedInternal(T data)
+    {
+        Console.Write(data);
+    }
+
+    internal static void OutputInternal(T data)
+    {
+        Console.Write(data);
+    }
+
+    private static void OutputPrivate(T data)
+    {
+        Console.Write(data);
+    }
+
+    public override bool Equals(object obj)
+    {
+        return base.Equals(obj);
+    }
+
+    public override int GetHashCode()
+    {
+        return base.GetHashCode();
+    }
+
+    public static bool operator ==(GenericType1<T> left, GenericType1<T> right)
+    {
+        return object.Equals(left, right);
+    }
+
+    public static bool operator !=(GenericType1<T> left, GenericType1<T> right)
+    {
+        return !object.Equals(left, right);
+    }
+}
+
+public class OpenType<T>
+{
+}
+
+public sealed class ClosedType : OpenType<String>
+{
+
+    public static void OutputProtected()
+    {
+    }
+
+    public static string Test
+    {
+        get { return string.Empty; }
+    }
+}");
+        }
+
+        [Fact]
+        public void Basic_CA1000_ShouldNotGenerate()
+        {
+            VerifyBasic(@"
+Public Class GenericType1(Of T)
+    Private Sub New()
+    End Sub
+
+    Shared Sub New()
+    End Sub
+
+    Protected Shared ReadOnly Property TestProtected() As String
+        Get
+            Return String.Empty
+        End Get
+    End Property
+
+    Protected Friend Shared ReadOnly Property TestProtectedInternal() As String
+        Get
+            Return String.Empty
+        End Get
+    End Property
+
+    Friend Shared ReadOnly Property TestInternal() As String
+        Get
+            Return String.Empty
+        End Get
+    End Property
+
+    Private Shared ReadOnly Property TestPrivate() As String
+        Get
+            Return String.Empty
+        End Get
+    End Property
+
+    Protected Shared Sub OutputProtected(data As T)
+        Console.Write(data)
+    End Sub
+
+    Protected Friend Shared Sub OutputProtectedInternal(data As T)
+        Console.Write(data)
+    End Sub
+
+    Friend Shared Sub OutputInternal(data As T)
+        Console.Write(data)
+    End Sub
+
+    Private Shared Sub OutputPrivate(data As T)
+        Console.Write(data)
+    End Sub
+
+    Public Overrides Function Equals(obj As Object) As Boolean
+        Return MyBase.Equals(obj)
+    End Function
+
+    Public Overrides Function GetHashCode() As Integer
+        Return MyBase.GetHashCode()
+    End Function
+
+    Public Shared Operator =(left As GenericType1(Of T), right As GenericType1(Of T)) As Boolean
+        Return Object.Equals(left, right)
+    End Operator
+
+    Public Shared Operator <>(left As GenericType1(Of T), right As GenericType1(Of T)) As Boolean
+        Return Not Object.Equals(left, right)
+    End Operator
+End Class
+
+Public Class OpenType(Of T)
+End Class
+
+Public NotInheritable Class ClosedType
+    Inherits OpenType(Of [String])
```

Comment:
:) I was wondering too because the [] escapes the keyword and makes it an identifier but since it's used as a type that will just bind to System.String again (as opposed to something like `Dim [String] As Integer = 10`)